src/field_theory/cardinality.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import ring_theory.localization.cardinality
+import algebra.is_prime_pow
+import field_theory.finite.galois_field
+import data.equiv.transfer_instance
+import algebra.ring.ulift
+import data.mv_polynomial.cardinal
+import data.rat.denumerable
+
+/-!
+# Cardinality of Fields
+
+In this file we show all the possible cardinalities of fields. All infinite cardinals can harbour
+a field structure, and so can all types with prime power cardinalities, and this is sharp.
+
+## Main statements
+
+* `fintype.nonempty_field_iff`: A `fintype` can be given a field structure iff its cardinality is a
+  prime power.
+* `infinite.nonempty_field` : Any infinite type can be endowed a field structure.
+
+-/
+
+local notation `‖` x `‖` := fintype.card x
+
+open_locale cardinal non_zero_divisors
+
+universe u
+
+/-- A finite field has prime power cardinality. -/
+lemma is_prime_pow_card_finite_field {α} [fintype α] [field α] : is_prime_pow (‖α‖) :=
```

Comment:
Arguably this should have a `fintype` in its name somewhere since it's about `fintype.card`, although I don't know what the convention in practice is.

Reply:
I feel like it's clear what's meant here? But maybe `fintype.is_prime_pow_card_of_field` is reasonable, too

Reply:
(or the other way round)